sdk/trace/batch_span_processor.go

```diff
@@ -326,11 +340,9 @@ func (bsp *batchSpanProcessor) drainQueue() {
 	for {
 		select {
 		case sd := <-bsp.queue:
-			if sd == nil {
-				if err := bsp.exportSpans(ctx); err != nil {
-					otel.Handle(err)
-				}
-				return
+			if _, ok := sd.(forceFlushSpan); ok {
+				// Ignore flush requests as they are not valid spans.
+				continue
```

Comment:
Fix for https://github.com/open-telemetry/opentelemetry-go/issues/4335. Another reason for https://github.com/open-telemetry/opentelemetry-go/issues/4354